### ISSUE TITLE
Messages don't get announced automatically to screen-reader users.

### ISSUE THREAD

**Issue by sideshowbarker** [gitterHQ/gitter#769]
Messages in gitter rooms don’t get announced automatically to users of AT software/screen-reader (JAWS, VoiceOver, etc.), which makes for a not-so-great user experience for those users.

My colleagues @ljwatson and @stevefaulkner can provide info and help with testing as needed.

**Comment by MichielBijl** [comment]
This does work with Safari and VoiceOver on OS X 10.10.

**Comment by LjWatson** [comment]
Confirmed that the notifications are not announced with either NVDA 2015.3 or Jaws 16, in Firefox 39, Chrome 43 or IE11.

**Comment by sideshowbarker** [comment]
> Confirmed that the notifications are not announced with either NVDA 2015.3 or Jaws 16, in Firefox 39, Chrome 43 or IE11.

Chrome and Safari are the only browsers this can/should be tested with at this point.

The reason is that IE11 doesn’t support Web Notifications at all yet, and Firefox currently has a bug https://bugzilla.mozilla.org/show_bug.cgi?id=1052776 which prevents it from causing Notifications to be announced/exposed to AT/screen-reader users at all.

**Comment by MadLittleMods** [comment]
Desktop notifications are announced properly with ChromeVox v45.0.2428.0 on Chrome 43.0.2357.134 but those notifications only pop up when the Gitter tab has focus.

Adding an [ARIA live-region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) should be sufficient. I did a little testing and the live region seems to work out of the box pretty well. ChromeVox currently doesn't read the author though.

```
<ul aria-live="polite" role="log"></ul>
```

**Comment by MadLittleMods** [comment]
This will be in the next release. Seems to perform fine with NVDA and ChromeVox.

A note on OSX(10.11.2) Voiceover: no matter what concoction of attributes or recipe I try to use, VoiceOver just doesn't play well. I would love some insight into making this work properly everywhere or if you know of an example somewhere that works, please share.

VoiceOver seems to always either announce the whole area when a change occurs, I would think the default or setting an explicit `aria-atomic` should be taking care of this but it doesn't seem to be respected. Or you can get it to announce only the new parts of the region, but this requires certain CSS to give it a height constraint of some sort and overflow. Even then, it starts going wonky once you hit the overflow.

I wrote up a standalone barebones tests: 

 - https://jsfiddle.net/MadLittleMods/r9pw5u9d/
 - http://codepen.io/MadLittleMods/pen/QyrrWb


---

Just some other demos and reference:

 - http://chat.karlgroves-sandbox.com/
 - https://github.com/kmoe/crypto-chat/issues/12 (crypto-chat doesn't perform properly)

.

 - https://www.w3.org./WAI/PF/aria-practices/#liveprops
 - http://juicystudio.com/article/wai-aria_live-regions_updated.php
